pycomlink/processing/processor.py

```diff
@@ -102,13 +104,55 @@ def __init__(self, cml):
                                   ['txrx', 'baseline'],
                                   'A')
 
+        self.calc_A_min_max = cml_wrapper(cml,
+                                          _calc_A_min_max,
+                                          ['tx_min', 'tx_max', 
+                                           'rx_min', 'rx_max'],
+                                          'Ar_max')
+
 
 # TODO: Integarte this somewhere else, since this
 #       sould be carried out after every baseline determination
 def _calc_A(txrx, baseline):
     return txrx - baseline
 
 
+def _calc_A_min_max(tx_min, tx_max, rx_min, rx_max, gT=1.0, gR=0.6, window=7):
+    """Calculate rain rate from attenuation using the A-R Relationship
+    Parameters
+    ----------
+    gT : float, optional
+        induced bias
+    gR : float, optional
+        induced bias
+    window: int, optional
+        number of previous measurements to use for zero-level calculation
+    Returns
+    -------
+    float or iterable of float
+        Ar_max
+    Note
+    ----
+    Based on: "Empirical Study of the Quantization Bias Effects in
+    Commercial Microwave Links Min/Max Attenuation
+    Measurements for Rain Monitoring" by OSTROMETZKY J., ESHEL A.
+    """
+
+    # quantization bias correction
+    Ac_max = tx_max - rx_min + (gT + gR) / 2
+    Ac_min = tx_min - rx_max - (gT + gR) / 2
+
+    Ac_max[np.isnan(Ac_max)] = np.rint(np.nanmean(Ac_max))
+    Ac_min[np.isnan(Ac_min)] = np.rint(np.nanmean(Ac_min))
+
+    # zero-level calculation
+    Ar_max = np.full(Ac_max.shape, 0)
+    for i in range(len(Ac_max)):
+        Ar_max[i] = Ac_max[i] - Ac_min[max(0, i-window):i+1].min()
```

Comment:
Wouldn't it be more correct to have NaN as a results at the beginning of the time series until the window can be fully filled with values from `Ac_min`?

Reply:
We prefer using any available data in order to avoid losing large time periods or rain.
For example, if the sampling period of 15 minutes and the window size is 8 - we can lose 2 hours of a rain event (at the beginning of the time series).